Tests for enum use in cutParser

Dr15Jones opened the pull request:
If I run with multiple threads in the ROOT6  we get an intermittent failure from the cutParser. This problem was traced back to TEnum not being thread safe. The test added by this pull request replicates the circumstances of the failure. This test will be useful for monitoring the thread safety of the cutParser for both ROOT 5 and 6.

Dr15Jones committed:
Add tests for usage of enums in the cutParser

Test for both success and failure when using enums with the cutParser.

Dr15Jones committed:
Test the cutParser usage with multiple threads

Tests using the threaded framework have shown failures using the
cutParser from multiple threads. This test replicates the circumstances
of the failure and if the planets all align can replicate the failure.

Dr15Jones commented:
@davidlange6 This is the same pull request as #7974 but for 7_4.

cmsbuild commented:
A new Pull Request was created by @Dr15Jones (Chris Jones) for CMSSW_7_4_X.

Tests for enum use in cutParser

It involves the following packages:

CommonTools/Utils


@nclopezo, @cvuosalo, @monttj, @cmsbuild, @slava77, @vadler can you please review it and eventually sign? Thanks.
You can sign-off by replying to this message having '+1' in the first line of your reply.
You can reject by replying  to this message having '-1' in the first line of your reply.
If you are a L2 or a release manager you can ask for tests by saying 'please test' in the first line of a comment.
@degano you are the release manager for this.
You can merge this pull request by typing 'merge' in the first line of your comment.

Dr15Jones commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
-1 
 
Tested at: 0e6cc758e200ae84b87b258a00acf15cd6e2f965
I ran the usual tests and I found errors in the following unit tests: 
 
 ---> test testCommonToolsUtilThreaded had ERRORS
---> test testCommonToolsUtil had ERRORS
 
 
you can see the results of the tests here: 
 https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7974/2958/summary.html

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7974/2958/summary.html

Dr15Jones commented:
I need slightly different code for ROOT 5 and ROOT 6 because RootAutoLibraryLoader was removed from our ROOT 6 branch.

Dr15Jones committed:
Made test work with ROOT 5

ROOT 5 needs Cintex enabled and the type of exception thrown when
there is an unknown enum is different.

cmsbuild commented:
Pull request #7974 was updated. @nclopezo, @cvuosalo, @monttj, @cmsbuild, @slava77, @vadler can you please check and sign again.

Dr15Jones commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: 42a60dec71d951db9a10651a896563dd4956434f
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7974/2978/summary.html

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7974/2978/summary.html

cvuosalo commented:
+1

For #7974 42a60dec71d951db9a10651a896563dd4956434f

This PR provides a test to monitor the thread safety of the cutParser for both ROOT 5 and 6.

The added code is satisfactory, and the Jenkins tests show no problems.

slava77 commented:
+1

trying to see if the bot listens to me
